akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -36,6 +36,40 @@ private[akka] final case class Filter[T](p: T ⇒ Boolean, decider: Supervision.
   override def decide(t: Throwable): Supervision.Directive = decider(t)
 }
 
+/**
+ * INTERNAL API
+ */
+private[akka] final case class TakeWhile[T](p: T ⇒ Boolean, decider: Supervision.Decider) extends PushStage[T, T] {
+  var taking = true
+
+  override def onPush(elem: T, ctx: Context[T]): SyncDirective =
+    if (taking && p(elem)) {
+      ctx.push(elem)
+    } else {
+      taking = false
+      ctx.pull
+    }
+
+  override def decide(t: Throwable): Supervision.Directive = decider(t)
+}
+
+/**
+ * INTERNAL API
+ */
+private[akka] final case class DropWhile[T](p: T ⇒ Boolean, decider: Supervision.Decider) extends PushStage[T, T] {
+  var dropping = true
+
+  override def onPush(elem: T, ctx: Context[T]): SyncDirective =
+    if (dropping && p(elem)) {
+      ctx.pull
+    } else {
+      dropping = false
+      ctx.push(elem)
+    }
```

Comment:
Looks ok, question is if we want to inverse the if or not ("dropping for a long time is the more often case or just dropping for a little bit is?")